tests/grammar/test_decorators_grammar.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+from pytest import (
+    raises,
+)
+
+from lark.exceptions import (
+    UnexpectedToken,
+)
+
+valid_decorator_codes = [
+    """
+@public
+def test() -> uint256:
+    return 123
+    """,
+    """
+@private
+def test() -> uint256:
+    return 123
+    """,
+    """
+@constant
+def test() -> uint256:
+    return 123
+    """,
+    """
+@nonreentrant("test")
+def test() -> uint256:
+    return 123
+    """
+]
+
+
+@pytest.mark.parametrize('good_code', valid_decorator_codes)
+def test_grammar_good_decorators(good_code, lark_grammar):
+    assert lark_grammar.parse(good_code + "\n")
```

Comment:
Are these sanity checks that the grammar is working?

Reply:
Yup.